File: topKElement.py
def topKElement(nums, k):

	count = {} # create counter dict
	freq = [[] for i in range(len(nums)+1)] # convert 

	for n in nums:

		count[n] = 1 + count.get(n, 0)

	print(count)

	for n, c in count.items():
		freq[c].append(n)

	print(freq)

	res = []
	for i in range(len(freq)-1, -1, -1):
		
		for n in freq[i]:
			res.append(n)

			if len(res) == k:
				return res


nums = [1,1,1,2,2,3]
k = 2
# print(topKElement(nums, k))
print([[]for i in range(len(nums))])

File: topKElement.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using std::vector;
using std::unordered_map;
using std::cout;
using std::endl;

class Solution {
public:
    
    vector<int> topKFrequent(vector<int>& nums, int k) {

    	int n = nums.size();
        
        unordered_map<int, int> count;
        std::vector<std::vector <int>> freq(n+1);
        std::vector <int> res;

        for (auto i : nums) {
            count[i] ++;
        }
        
        
        for (auto item : count){
			freq[item.second].push_back(item.first);            
        }

        for (int i=n; i>=0; i--){

        	for (auto j : freq[i]){
        		res.push_back(j);

        		if (res.size() == k) 
        			return res;
        	}
        }
        
    }
};


int main()
{
	vector<int> nums {1,1,1,2,2,3};
	int k = 2;

	Solution s;

	std::cout << s.topKFrequent(nums, k) << std::endl;
	return 0;
}